### ISSUE TITLE
High memory usage, possible memory leaks

### ISSUE THREAD

**Issue by Medvidas** [daveake/HABBase#15]
Hi Dave,
The HABBase application gradually occupies an enormous amount of memory.
I am running the 32bit, v1.7.2, Win 10 22H2. 
After starting it asks less than 200 MB, but it takes up practically 2GB of RAM after 6 hours.

Regards, Ales

![habhub leaks](https://github.com/daveake/HABBase/assets/57909431/307ff935-d539-4db6-87a1-42dd12c0c97f)

**Comment by daveake** [comment]
Can you tell me please:

- which data source(s) you've set up
- which of those are receiving telemetry
- which of those are receiving SSD
- which upload target(s) are enabled

Thanks

**Comment by Medvidas** [comment]
Hi

my HABBase was updated from v1.6.8 approx. 2 weeks ago.
I am using Sondehub pooling and Lora sources, nothing fancy.

The app was reinstalled yesterday from scratch with Sondehub pooling only.
It crashed today – see attachment.
It consumes approx. 6 MB per minute, with no other sources or upload, mostly minimized in the tray.
As HB was out of memory and killed in the task manager, it was not possible to start it again – see attachment.
The likely cause was the Settings.json zeroed! Without backup I reinstalled the app again 😊
Idea: Once save in settings is initiated, old settings.json is renamed/copied to *.bak

One more thing related to another issue – once I deleted all the data sources no ADD SOURCE BUTTON was displayed.
It was necessary to restart the app, then the button was visible.

Hope it will help.
Wish you the best, Ales
![hb3](https://github.com/daveake/HABBase/assets/57909431/72246f8b-9d2d-4f67-85ab-891dfd6c819f)
![hbase2](https://github.com/daveake/HABBase/assets/57909431/1cd104b4-6f73-4adb-bb3b-feaaaf50b52e)
